### PR TITLE
adds a search box to the toolbar featureflag list

### PR DESCRIPTION
**Pull request by pauldambra**
## Changes

When there are lots of feature flags it is hard to find flags in the toolbar feature flags list.

This adds a search box that filters the list of flags

feature-flag-list-search

## How did you test this code?

by running it locally

attempting to write a test case for the feature flag logic led to a test runner error

```
/Users/pauldambra/github/posthog/node_modules/query-selector-shadow-dom/src/querySelectorDeep.js:6
    import { normalizeSelector } from './normalize';
    ^^^^^^

    SyntaxError: Cannot use import statement outside a module
```

So I didn't write any tests

### CODE REVIEW

**Review by mariusandra — CHANGES_REQUESTED** [review]
Love the addition of this, though I'd turn the logic around. Comment inline.

**Review by mariusandra — CHANGES_REQUESTED** [review]
I patched the issue jest had with this logic. JS can be fun 😅.

I'd still like you to rework the selectors here in the way I suggested before. 

Imagine data flowing through the app as a river, starting with bytes on disk, ending with your user's eyeballs:

image

There's a way to make this river flow smoothly through Kea (in a way some would call *functional programming*), but the current implementation adds a few pumps and loops to move the water around.

Basically, instead of using a lookup table to filter results, make the lookup table itself the filtered results... 🤔

**Comment by mariusandra** [comment]
Hm, again, why are you filtering out names, sending them to Fuse, and later using a set-based lookup table in yet another filter? Just send the entire flags array to fuse and let it remove those where the name doesn't match.

**Comment by pauldambra** [comment]
> Just send the entire flags array to fuse and let it remove those where the name doesn't match.

Ah, ok, went away and read the Fuse docs. I _think_ I understand now

**Review by mariusandra — APPROVED** [review]
This looks good now! Feel free to do something with the tests as well, and then merge it in.

**Comment by mariusandra** [comment]
This issue is very closely related though... Copying the skittish from here should be an easy fix to whoever wants to take this on
https://github.com/PostHog/posthog/issues/6202
